test/test_tensorflow.py

```diff
@@ -975,6 +975,38 @@ def test_horovod_broadcast_grad_gpu(self):
                             "gradient %s differs from expected %s, "
                             "error: %s" % (grad_out, expected, str(err)))
 
+    def test_horovod_broadcast_eager_mode_error(self):
+        """Test that tries to broadcast tensorflow global variables
+        in eager execution mode. This call should raise a RuntimeError."""
+
+        if not hvd.util._has_eager:
```

Comment:
These tests are already executing eagerly and in graph mode (can be checked via `executing_eagerly()`), maybe we can just check for that and skip the test if there is a mode mismatch?

Reply:
Oh that explain why I had so much trouble allowing/disabling eager exec. I'll update my code :)